src/nvim/eval/funcs.c

```diff
@@ -3386,15 +3386,17 @@ static void f_getcompletion(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     emsg(_(e_invarg));
     return;
   }
+  const char *pattern = tv_get_string(&argvars[0]);
```

Comment:
The blank line below this line should probably be moved above this line.

Reply:
It follows what happens above:
```
  if (argvars[1].v_type != VAR_STRING) {
    semsg(_(e_invarg2), "type must be a string");
    return;
  }
  const char *const type = tv_get_string(&argvars[1]);
```

Reply:
Oh, in the Vim patch it is above that block.

Reply:
In the vim patch `argvars[0].v_type` is not checked and I wasn't sure if that's correct, so I moved it after the type check.

Reply:
It is actually checked above, but that check probably depends on other patches.

Reply:
One thing about this line, Bram reuses `char_u *pat`, but `tv_get_string` returns a `const char*`. Should I cast away the const and do the same thing?

Reply:
I'm not sure either.

Reply:
For what it's worth, const is already casted away in `xpc.xp_pattern = (char_u *)pat;`, line 3399.